samples/sample_script.sh

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env sh
+
+# set -xv
+
+# show usage and quit
+function usage()
+{
+  printf "usage: %s <command>\n" $(basename $0) >&2
+  printf "\n" >&2
+  printf "    commands are:\n"
+  printf "        download [<version>] - downloads and mounts all example data\n" >&2
+  printf "                               gets latest version unless <version> supplied\n" >&2
+  printf "        remove - removes all example data\n" >&2
+  printf "        version [<version>] - get version <version>\n" >&2
+  printf "                            - without a parameter, shows current the version\n" >&2
+  printf "        versions - list available versions\n" >&2
+  exit 2
+}
+
+
+# complain and quit
+function fail_out()
+{
+  printf "Failed! %s\n" "$@" >&2
+  exit 2
+}
+
+
+# check that we have the expected enlistment directory; download it if not
+function ensure_enlistment()
+{
+  if [ ! -d $target_path/.git ]; then
+     printf "no examples collection at %s; dowloading ..." $target_path >&2
+     do_download
+  fi
+}
+
+
+# clone the git repo, don't report progress but DO report errors
+function do_download()
+{
+  if [ -d $target_path/.git ]; then
+    printf "examples collection already exists at %s\n" $target_path >&2
```

Comment:
I've added a pull here. I didn't do initially because, if the user has any changes to the files, then things can get complicated with git and the script would need to handle a lot of different states.